core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -2871,6 +2876,31 @@ public void testRegisterExistingTable() {
     assertThat(catalog.dropTable(identifier)).isTrue();
   }
 
+  @Test
+  public void testRegisterAndOverwriteExistingTable() {
+    Assumptions.assumeTrue(supportsRegisterTableWithOverwrite());
+    C catalog = catalog();
+
+    TableIdentifier identifier = TableIdentifier.of("a", "t1");
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(identifier.namespace());
+    }
+
+    catalog.createTable(identifier, SCHEMA);
```

Comment:
nit: In CatalogTests I see that different logical blocks within the tests are separated by empty lines.